custom_components/hikvision_next/notifications.py

```diff
@@ -52,60 +60,127 @@ async def post(self, request: web.Request):
         )
         return response
 
-
-def trigger_sensor(hass: core.HomeAssistant, xml: str) -> None:
-    """Determine entity and set binary sensor state"""
-
-    alert = ISAPI.parse_event_notification(xml)
-
-    device_serial = alert.device_serial
-
-    if not device_serial and alert.mac:
-        # get device_serial by mac
+    async def parse_event_request(self, request: web.Request) -> str:
+        """Extract XML content from multipart request or from simple request"""
+
+        data = await request.read()
+
+        content_type_header = request.headers.get(CONTENT_TYPE)
+
+        _LOGGER.debug("request headers: %s", request.headers)
+        xml = None
+        if content_type_header in CONTENT_TYPE_XML:
+            xml = data.decode("utf-8")
+        else:
+            # "multipart/form-data; boundary=boundary"
+            decoder = MultipartDecoder(data, content_type_header)
+            for part in decoder.parts:
+                headers = {}
+                for key, value in part.headers.items():
+                    assert isinstance(key, bytes)
+                    headers[key.decode("ascii")] = value.decode("ascii")
+                _LOGGER.debug("part headers: %s", headers)
+                if headers.get(CONTENT_TYPE) in CONTENT_TYPE_XML:
+                    xml = part.text
+                if headers.get(CONTENT_TYPE) == CONTENT_TYPE_IMAGE:
+                    _LOGGER.debug("image found")
+
+        if not xml:
+            raise ValueError(f"Unexpected event Content-Type {content_type_header}")
+        return xml
+
+    def parse_event_notification(self, xml: str) -> AlertInfo:
```

Comment:
because parse_event_notification contains keys related to ISAPI  better 
keep it in isapi.py